### PR TITLE
[MM-46818] Ask for email on confirmation popin

### PR DESCRIPTION
**Pull request by JulienTant**
#### Summary
Ask for an optional email address on the confirmation pop in. I validated with Don that it was ok for him.

#### Screenshot
image

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-46818

### CODE REVIEW

File: webapp/src/components/confirm_feedback_modal/confirm_feedback_modal.jsx [hunk 1]
```diff
@@ -59,4 +83,4 @@ export default class ConfirmFeedbackModal extends React.PureComponent {
             </Modal>
         );
     }
-}
\ No newline at end of file
+}
```
Comment (pvev) [inline webapp/src/components/confirm_feedback_modal/confirm_feedback_modal.jsx:86]: blank line at eof

**Review by pvev — APPROVED** [review]
LGTM! thanks @JulienTant :shipit:

**Comment by JulienTant** [comment]
Hi @annelieseklein and @justinegeffen

I need some feedback/validation on the text+UX of this modal. FYI before my changes, the modal just had the first line of text, this PR adds the second line + the text field.

Thank you!!

File: webapp/src/components/confirm_feedback_modal/confirm_feedback_modal.jsx [hunk 2]
```diff
@@ -37,7 +50,18 @@ export default class ConfirmFeedbackModal extends React.PureComponent {
                     <Modal.Title>{'Send Feedback'}</Modal.Title>
                 </Modal.Header>
                 <Modal.Body>
-                    {'You are about to send feedback about Mattermost. Are you sure?'}
+                    <p>{'You are about to send feedback about Mattermost. Are you sure?'}</p>
+                    <p>{'You can optionally provide an email address if you wish to be contacted by our team.'}</p>
```
Comment (justinegeffen) [inline webapp/src/components/confirm_feedback_modal/confirm_feedback_modal.jsx:54]:
```suggestion
                    <p>{'**Optional**: If you'd like to be contacted by our team, share your your email address.'}</p>
```

File: webapp/src/components/confirm_feedback_modal/confirm_feedback_modal.jsx [hunk 3]
```diff
@@ -37,7 +50,18 @@ export default class ConfirmFeedbackModal extends React.PureComponent {
                     <Modal.Title>{'Send Feedback'}</Modal.Title>
                 </Modal.Header>
                 <Modal.Body>
-                    {'You are about to send feedback about Mattermost. Are you sure?'}
+                    <p>{'You are about to send feedback about Mattermost. Are you sure?'}</p>
```
Comment (justinegeffen) [inline webapp/src/components/confirm_feedback_modal/confirm_feedback_modal.jsx:53]:
```suggestion
                    <p>{'Send us your feedback and suggestions.'}</p>
```

File: webapp/src/components/confirm_feedback_modal/confirm_feedback_modal.jsx [hunk 4]
```diff
@@ -37,7 +50,18 @@ export default class ConfirmFeedbackModal extends React.PureComponent {
                     <Modal.Title>{'Send Feedback'}</Modal.Title>
```
Comment (justinegeffen) [inline webapp/src/components/confirm_feedback_modal/confirm_feedback_modal.jsx:50]:
```suggestion
                    <Modal.Title>{'Send feedback'}</Modal.Title>
```

**Comment by justinegeffen** [comment]
Thanks, @JulienTant! Is it possible to generate an example of the modal with my suggestions? I'd like to make sure it reads well.

File: webapp/src/components/confirm_feedback_modal/confirm_feedback_modal.jsx (hunk 3 above)
Comment (JulienTant) [reply webapp/src/components/confirm_feedback_modal/confirm_feedback_modal.jsx:53]: @justinegeffen at this stage the user already typed his feedback and sent the message. We are trying to confirm that they are indeed sure they want to send it

**Comment by JulienTant** [comment]
@justinegeffen So here's a screenshot with your changes (excluding the first phrase one). The way it works is the user types a feedback (red circle) and when they try to send it, then the modal appears to confirm they want to send the message

image

File: webapp/src/components/confirm_feedback_modal/confirm_feedback_modal.jsx (hunk 3 above)
Comment (justinegeffen) [reply webapp/src/components/confirm_feedback_modal/confirm_feedback_modal.jsx:53]: Ah! Gotcha. Ok, so then let's go with:

"You're about to send feedback about Mattermost."

Comment (JulienTant) [reply webapp/src/components/confirm_feedback_modal/confirm_feedback_modal.jsx:53]: Without the "Are you sure?"

**Comment by JulienTant** [comment]
Alright here's the last version: 
image

**Comment by annelieseklein** [comment]
"Confirm" also works. Instead of "If you'd like to be contacted by our team...." can we say something like "If you're open to being contacted for research purposes, please include your email."
@JulienTant

**Comment by justinegeffen** [comment]
Good point, @JulienTant! I'm happy with "Confirm" to replace "Yes" and I've added @annelieseklein's excellent suggested copy inline (not sure I got the formatting correct which is why I didn't commit it).

**Comment by JulienTant** [comment]
Updated

image

**Comment by justinegeffen** [comment]
This looks awesome, thank you!

**Review by justinegeffen — APPROVED** [review]

**Pull request merged**